modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -498,6 +539,10 @@ private CompletableFuture<Void> handle(VersionedUpdate update, HybridTimestamp m
         }
     }
 
+    private static CatalogZoneDescriptor tableZoneDescriptor(Catalog oldCatalog, int tableId) {
+        return Objects.requireNonNull(oldCatalog.zone(Objects.requireNonNull(oldCatalog.table(tableId)).zoneId()));
```

Comment:
such pattern doesn't make any sense. You either have to provide context for every `requireNonNull` (like `Objects.requireNonNull(oldCatalog.table(tableId), "table")`), or put `requireNonNull` on distinct line with temporal variable (`var table = Objects.requireNonNull(oldCatalog.table(tableId))`). Otherwise you will get `NPE at line X` with no chance to get what exactly was not found in catalog